build/dsc/templates/dsc_type_spec.rb.erb

```diff
@@ -44,7 +44,17 @@ describe Puppet::Type.type(:dsc_<%= resource.friendlyname.downcase %>) do
 <%    end # if property.required? -%>
 <%    if property.values -%>
 <%      property.values.each do |value| -%>
+<%    if property.array? -%>
```

Comment:
You'll also want to reinsert a newline prior to this one to preserve previous whitespace.